static/js/components.js

```diff
@@ -0,0 +1,110 @@
+class YoutubePlayer extends HTMLElement {
+    constructor() {
+        super();
+    }
+
+    connectedCallback() {
+        const shadow = this.attachShadow({ mode: "open" });
+        const style = document.createElement("style");
+        style.textContent = `
+            .youtube_placeholder {
```

Comment:
```suggestion
        style.textContent = `
            iframe {
                aspect-ratio: 16/9;
                width: 100%;
            }
            .youtube_placeholder {
```

Fix for the scaling issue of the iframe. Seems like outer CSS will NOT affect webcomponents.

Reply:
I had completely missed the external `iframe` CSS, thanks!